include/quiche.h

```diff
@@ -270,6 +270,16 @@ bool quiche_conn_is_in_early_data(quiche_conn *conn);
 // Returns true if the connection is closed.
 bool quiche_conn_is_closed(quiche_conn *conn);
 
+// Initializes the stream's application data.
+//
+// Note that the application is responsible for freeing the data.
+int quiche_conn_stream_init_application_data(quiche_conn *conn,
+                                             uint64_t stream_id,
+                                             void *data);
+
+// Returns the stream's application data, if any was initialized.
+void *quiche_conn_stream_application_data(quiche_conn *conn, uint64_t stream_id);
```

Comment:
This looks exactly what I want. If I'd bikeshed anything on this, it would perhaps be that the function names could be called *set* and *get* stream application-data, but I don't care strongly.